### PR TITLE
Add certified nodes list page

### PR DESCRIPTION
**Pull request by Yndira-FlowForge**
## Description

Added a page for our certified nodes

## Related Issue(s)

closes [#1257](https://github.com/FlowFuse/website/issues/1257)

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowforge/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by Yndira-FlowForge** [comment]
- I've added the page with the list, and you can filter it by category.
- Each category has a brief description; feel free to edit it as needed.
- It's currently set up for one page per package, but I'm uncertain if we'll copy the content/structure from flows.nodered, pull it directly from that repo, or create new content around it.
- Currently, if we don't add content to the .md, it will link to the corresponding flows.nodered page; if we do add content, it will link to a page on our site.
- I've included it as the 7th reason in /why-flowfuse and linked it from there.

**Comment by Yndira-FlowForge** [comment]
The last two items with a green background didn't have a category or link, so I didn't include those.

Screenshot 2023-11-16 at 18 01 44

File: src/_data/certifiedCategories.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "value": "database"
+}, {
+    "label": "FlowFuse",
+    "value": "flowfuse-node"
+}, {
+    "label": "Formatting",
+    "value": "formatting"    
+}, {
+    "label": "Protocoll",
+    "value": "protocoll"
```
Comment (ZJvandeWeg) [inline src/_data/certifiedCategories.json:18]:
```suggestion
    "label": "Protocol",
    "value": "protocol"    
```

File: src/_data/reasons.json [hunk 2]
```diff
@@ -25,4 +25,9 @@
   {
     "title": "Remote device management",
     "content": "Many people are using Node-RED to aggregate and analyze data from remote devices. However, Node-RED is really meant for running locally or in the cloud. FlowFuse breaks down the barriers to remote device management by providing a central platform that can manage hundreds of devices running Node-RED. A great way to scale up your use of Node-RED in your organization."
+  }
+  ,
+  {
```
Comment (ZJvandeWeg) [inline src/_data/reasons.json:30]:
```suggestion
  }, {
```

File: src/certified-nodes/communication/email.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (ZJvandeWeg) [inline src/certified-nodes/communication/email.md:1]: Can we create pages from data? I feel like the JSON (or better yet; YAML) should be stored instead of a markdown with frontmatter.

File: src/product/why-flowfuse.njk
Comment (ZJvandeWeg) [inline src/product/why-flowfuse.njk:1]: This change should be a different PR next time

File: src/certified-nodes/protocoll.njk
Comment (ZJvandeWeg) [inline src/certified-nodes/protocoll.njk:1]: Single L in protocol. :) Please update the file name and directory name

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/certified-nodes/communication/email.md (hunk 3 above)
Comment (Yndira-FlowForge) [reply src/certified-nodes/communication/email.md:1]: You mentioned [here](https://github.com/FlowFuse/website/issues/1257#issuecomment-1804206455) that:

> I think a list and a page per package would be great, though as first iteration a list will suffice.

Is this still the plan? That's why I added all the md files, to have a page per package in place. And I had some questions about it too, [here](https://github.com/FlowFuse/website/pull/1323#issuecomment-1814856726).

Comment (ZJvandeWeg) [reply src/certified-nodes/communication/email.md:1]: @Yndira-FlowForge Please read: https://www.11ty.dev/docs/pages-from-data/

Comment (Yndira-FlowForge) [reply src/certified-nodes/communication/email.md]: Thanks for the link :relaxed: I read it and put it into practice, but it only creates html files in the output folders, there's no data that I can use for filtering by category or even display anything from them, and we still would have empty pages. Maybe you have a better idea of how this could be useful that I'm not seeing?

I implemented a different approach that doesn't create any pages, just the list and links to the flows.nodered page. I think as a first iteration it works.

Comment (Yndira-FlowForge) [reply src/certified-nodes/communication/email.md]: @ZJvandeWeg, I believe I understand it now after examining some of the core-nodes files. I'll continue iterating, but I think we should ship it as is for our first iteration. What do you think?

File: src/_data/certifiedNodes.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+[{
```
Comment (ZJvandeWeg) [inline src/_data/certifiedNodes.json:1]: Next time let's construct this from the remote repository kind of like how we include the docs, or the blueprints

File: src/_data/certifiedNodes.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "description": "Plugin that allows you to interact with the FlowForge platform.",
+    "tags": ["flowfuse-node"]
+  },{
+    "title": "Base64 Formatter",
+    "id": "node-red-node-base64",
+    "url": "https://flows.nodered.org/node/node-red-node-base64",
+    "description": "Node to pack and unpack objects to base64 format",
+    "tags": ["formatting"]
+  },{
+    "title": "Buffer Parserr",
```
Comment (ZJvandeWeg) [inline src/_data/certifiedNodes.json:38]:
```suggestion
    "title": "Buffer Parser",
```

File: src/_includes/certified/common.njk [hunk 6]
```diff
@@ -0,0 +1,29 @@
+    <div class="px-2 flex items-center gap-12">
```
Comment (ZJvandeWeg) [inline src/_includes/certified/common.njk:1]: `common.njk` I think would be better named like the actual function in the page. It's not quite a header, but maybe call it something else?

File: src/_includes/certified/template-all.njk [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<div class="ff-blog container m-auto text-left max-w-4xl pt-8 pb-12 w-full ff-full-bg">
```
Comment (ZJvandeWeg) [inline src/_includes/certified/template-all.njk:1]: template-list.njk might be better? or `template-overview.njk`?

File: src/_includes/certified/template-category.njk [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<div class="flex flex-col justify-between ff-blog container m-auto text-left max-w-4xl pt-8 pb-12 w-full ff-full-bg">
+    <div>
+    {% include "certified/common.njk" %}
+
+    <ul class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-4 px-2 mt-6">
+        {%- asyncEach node in certifiedNodes -%}
+        {% if page.url.includes('/' + node.tags + '/')%}
+            {% include "certified/certified.njk" %}
+        {% endif %}
```
Comment (ZJvandeWeg) [inline src/_includes/certified/template-category.njk:9]: This feels like a hacky solution to me, do you know how to clean it up? E.g. collections in the eleventy.js?

File: src/certified-nodes/database.njk [hunk 9]
```diff
@@ -0,0 +1,15 @@
+---
+layout: nohero
+pagination:
+  data: certifiedNodes
```
Comment (ZJvandeWeg) [inline src/certified-nodes/database.njk:4]: Shouldn't this work like:

```suggestion
  data: certifiedNodes.database
```

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/certified-nodes/database.njk (hunk 9 above)
Comment (Yndira-FlowForge) [reply src/certified-nodes/database.njk:4]: I added the Category key

File: src/_includes/certified/template-category.njk (hunk 8 above)
Comment (Yndira-FlowForge) [reply src/_includes/certified/template-category.njk]: I added the filter to eleventy.js and now a single template.njk works as a template for all, filtered and unfiltered.

**Comment by ZJvandeWeg** [comment]
@Yndira-FlowForge Can you follow up with pages for each node? Like: https://hub.meltano.com/loaders/target-airtable
